### PR TITLE
Add missing Dreamer connection; Fix index

### PR DESCRIPTION
**Pull request by rmorgens**
Add missing Dream mystic Connection, and fix Connection index missing MANY connection in the Compendium

### CODE REVIEW

File: src/items/class-features/dream_sending_(su).json [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    },
+    "recharge": {
+      "charged": false,
+      "value": null
+    },
+    "requirements": "12th Level (Mystic)",
+    "rollNotes": "",
+    "save": {
+      "type": "",
+      "dc": null,
```
Comment (LebombJames) [inline src/items/class-features/dream_sending_(su).json:101]: Add the DC for this ability

File: src/items/class-features/fortified_imagination_(su).json [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    "damage": {
+      "parts": []
+    },
+    "damageNotes": "",
+    "description": {
+      "chat": "",
+      "gmnotes": "",
+      "short": "",
+      "unidentified": "",
+      "value": "<p>&gt;@UUID[Compendium.sfrpg.class-features.udtZxolGt5mMq9SM]{Dreamer}</p><p class=\"MsoNormal\">You’ve experienced countless nightmares, preparing you to handle anything. You gain a +2 insight bonus to Will saves against mind-affecting spells and abilities. As a swift action, you can spend 1 Resolve Point to grant this bonus to an ally who can perceive you.</p>"
```
Comment (LebombJames) [inline src/items/class-features/fortified_imagination_(su).json:40]: MsoNormal

File: src/items/class-features/perceive_dreamspace_(su).json [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
...cal realm unique to each dreamer with whom you interact and explore with any senses you possess, though you can’t directly communicate with or affect the dreamer. Entering dreamspace is a full action. Once per day while inside dreamspace, you can spend 10 minutes concentrating on a single dreamer with whom you’re already familiar or a specific dream motif.</p><p class=\"MsoNormal\">If you observe a dreamer in dreamspace, the target must succeed at a Will save with a DC equal to 10 + your class level to resist, causing you to learn nothing useful. Otherwise, you learn 1d4 pieces of useful information about the dreamer at the GM’s discretion.</p><p class=\"MsoNormal\">If you observe a dream motif, you return to the waking world with the ability to create a physical manifestation of the motif as a standard action. This can mimic the effects of a 1st-level spell from any spell list. You must manifest the dream motif within 24 hours or until you next enter dreamspace (whichever comes first), or it’s lost.</p><p class=\"MsoNormal\">While inside dreamspace, you can’t perceive or interact with the physical world. Returning your consciousness fully to your physical body takes 1 minute.</p>"
```
Comment (LebombJames) [inline src/items/class-features/perceive_dreamspace_(su).json:40]: Several MsoNormals

File: src/items/class-features/perceive_dreamspace_(su).json [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+    },
+    "recharge": {
+      "charged": false,
+      "value": null
+    },
+    "requirements": "1st Level (Mystic)",
+    "rollNotes": "",
+    "save": {
+      "type": "",
+      "dc": null,
```
Comment (LebombJames) [inline src/items/class-features/perceive_dreamspace_(su).json:101]: Include the DC

**Review by LebombJames — CHANGES_REQUESTED** [review]
Go ahead and run cook again too.

File: src/items/class-features/dream_sending_(su).json (hunk 1 above)
Comment (rmorgens) [reply src/items/class-features/dream_sending_(su).json:101]: I can find no setting in Foundry for how to set that value...Please advise

File: src/items/class-features/perceive_dreamspace_(su).json (hunk 4 above)
Comment (LebombJames) [reply src/items/class-features/perceive_dreamspace_(su).json:106]: At the very bottom of the details tab, there's a saving throw and skill check field. Put the field there. May require a saving throw/skill check activation type. That is the preferred way to do saving throws, not inline formulas showing DCs

File: src/items/class-features/dream_sending_(su).json (hunk 1 above)
Comment (LebombJames) [reply src/items/class-features/dream_sending_(su).json:106]: At the very bottom of the details tab, there's a saving throw and skill check field. Put the field there. May require a saving throw/skill check activation type. That is the preferred way to do saving throws, not inline formulas showing DCs

File: src/items/class-features/perceive_dreamspace_(su).json (hunk 4 above)
Comment (LebombJames) [reply src/items/class-features/perceive_dreamspace_(su).json]: You've still got the saves in the wrong field. Check other class features with saving throws for examples

**Comment by rmorgens** [comment]
Looks like I finally got it correct.  Took a while to find an example.....

File: src/items/class-features/perceive_dreamspace_(su).json (hunk 4 above)
Comment (rmorgens) [reply src/items/class-features/perceive_dreamspace_(su).json]: I think I got it correct now

Comment (LebombJames) [reply src/items/class-features/perceive_dreamspace_(su).json]: Take out the inline formulas for the DCs as well

Comment (rmorgens) [reply src/items/class-features/perceive_dreamspace_(su).json]: Fixed

File: src/items/class-features/dream_sending_(su).json [hunk 5]
```diff
@@ -37,7 +37,7 @@
...
+      "value": "<p>&gt;@UUID[Compendium.sfrpg.class-features.udtZxolGt5mMq9SM]{Dreamer}</p>\n<p>Once per day while inside dreamspace, you can briefly communicate with creatures at great distances. This creature can be anywhere in the universe, including on another plane, as long as you’re familiar with them. You can speak freely with any ally linked by your @UUID[Compendium.sfrpg.class-features.ox6Am16XZqBXJNT9]{Telepathic Bond (Sp)} class feature when creating a dream sending. Creatures that aren’t linked to you by a telepathic bond can attempt a @Check[type:will save] with a DC equal to 15 + one-half your class level to resist the dream sending. A creature can choose to fail this save. On a failure, the creature experiences a dream containing a brief message of 25 words or less the next time it rests. The message can be verbal or nonverbal, and you control the aesthetic, imagery, and events of this dream. The creature wakes remembering your message, though you can’t influence a creature’s actions or attitude toward you using this ability.</p>"
```
Comment (LebombJames) [inline src/items/class-features/dream_sending_(su).json:40]: Just `@Check[type:will]`

File: src/items/class-features/perceive_dreamspace_(su).json [hunk 6]
```diff
@@ -37,7 +37,7 @@
...
...a physical realm unique to each dreamer with whom you interact and explore with any senses you possess, though you can’t directly communicate with or affect the dreamer. Entering dreamspace is a full action. Once per day while inside dreamspace, you can spend 10 minutes concentrating on a single dreamer with whom you’re already familiar or a specific dream motif.</p>\n<p>If you observe a dreamer in dreamspace, the target must succeed at a @Check[type:will] save with a DC equal to 10 + your class level to resist, causing you to learn nothing useful. Otherwise, you learn [[/r 1d4]]{1d4 (Perceive Dreamspace)} pieces of useful information about the dreamer at the GM’s discretion.</p>\n<p>If you observe a dream motif, you return to the waking world with the ability to create a physical manifestation of the motif as a standard action. This can mimic the effects of a 1st-level spell from any spell list. You must manifest the dream motif within 24 hours or until you next enter dreamspace (whichever comes first), or it’s lost.</p>\n<p>While inside dreamspace, you can’t perceive or interact with the physical world. Returning your consciousness fully to your physical body takes 1 minute.</p>"
```
Comment (LebombJames) [inline src/items/class-features/perceive_dreamspace_(su).json:40]: What's with `{1d4 (Perceive Dreamspace)}`? Why is there the brackets?

**Review by LebombJames — CHANGES_REQUESTED** [review]

File: src/items/class-features/visitation_(su).json [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+    "damage": {
+      "parts": []
+    },
+    "damageNotes": "",
+    "description": {
+      "chat": "",
+      "gmnotes": "",
+      "short": "",
+      "unidentified": "",
+      "value": "<p>&gt;@UUID[Compendium.sfrpg.class-features.udtZxolGt5mMq9SM]{Dreamer}</p>\n<p>Once per day, you can spend 1 Resolve Point as a full action to physically travel to the location of a specific dreamer through dreamspace. The target dreamer can be anywhere in the universe, including on another plane, as long as you’re familiar with them. The visitation lasts for a number of minutes equal to [[floor( @classes.mystic.levels / 2 )]]{one-half your character level}.</p>"
```
Comment (LebombJames) [inline src/items/class-features/visitation_(su).json:40]: This feature still has an inline DC, and is missing the DC in the proper place

File: src/items/class-features/fortified_imagination_(su).json [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+    "damage": {
+      "parts": []
+    },
+    "damageNotes": "",
+    "description": {
+      "chat": "",
+      "gmnotes": "",
+      "short": "",
+      "unidentified": "",
+      "value": "<p>&gt;@UUID[Compendium.sfrpg.class-features.udtZxolGt5mMq9SM]{Dreamer}</p>\n<p>You’ve experienced countless nightmares, preparing you to handle anything. You gain a +2 insight bonus to @Check[type:will] saves against mind-affecting spells and abilities. As a swift action, you can spend 1 Resolve Point to grant this bonus to an ally who can perceive you.</p>"
```
Comment (LebombJames) [inline src/items/class-features/fortified_imagination_(su).json:40]: The `@Check` here is not needed; it is simply mentioning Will saves, not calling for one

File: src/items/class-features/dreamer.json [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
...thetic, imagery, and events of this dream. The creature wakes remembering your message, though you can’t influence a creature’s actions or attitude toward you using this ability.</p><h2>@UUID[Compendium.sfrpg.class-features.FkLyMHbGPGwIsPJA]{Dream Blessing (Su) - 15th Level}</h2><p>Once per day, you can send a dream blessing to a creature with which you’re familiar. You control the aesthetic, imagery, and events of this dream. The creature experiences the dream when they next rest. For the next 24 hours, the creature benefits from a +2 bonus to a saving throw or skill check of your choice (this choice must be related to the dream’s subject matter) and can reroll a failed check of that type once per day</p><h2>@UUID[Compendium.sfrpg.class-features.HqBLb5H1BN81uHuB]{Visitation (Su) - 18th Level}</h2><p>Once per day, you can spend 1 Resolve Point as a full action to physically travel to the location of a specific dreamer through dreamspace. The target dreamer can be anywhere in the universe, including on another plane, as long as you’re familiar with them. The visitation lasts for a number of minutes equal to [[floor( @classes.mystic.levels / 2 )]]{one-half your character level}.</p>"
```
Comment (LebombJames) [inline src/items/class-features/dreamer.json:40]: Still a bunch of inline DCs here

**Review by LebombJames — CHANGES_REQUESTED** [review]

**Comment by rmorgens** [comment]
Should be ready for re-review

**Review by ian612 — DISMISSED** [review]

**Review by ian612 — COMMENTED** [review]
Just a few minor things to fix up left, mostly formatting for inline links and DCs

File: src/items/class-features/energy_reflection_(ex).json [hunk 10]
```diff
@@ -129,6 +130,10 @@
       "dc": "10+floor(@classes.mechanic.levels/2) + (@classes.mechanic.keyAbilityMod)",
       "descriptor": "half"
     },
+    "skillCheck": {
+      "type": "",
+      "dc": ""
+    },
     "source": "IS, pg. 25",
```
[the remark this answers is not in the document]
Comment (rmorgens) [reply src/items/class-features/energy_reflection_(ex).json:137]: This file shouldn't be in the PR

File: src/items/class-features/dreamer.json [hunk 11]
```diff
@@ -0,0 +1,127 @@
...
...agery, and events of this dream. The creature wakes remembering your message, though you can’t influence a creature’s actions or attitude toward you using this ability.</p><h2>@UUID[Compendium.sfrpg.class-features.Item.FkLyMHbGPGwIsPJA]{Dream Blessing (Su) - 15th Level}</h2><p>Once per day, you can send a dream blessing to a creature with which you’re familiar. You control the aesthetic, imagery, and events of this dream. The creature experiences the dream when they next rest. For the next 24 hours, the creature benefits from a +2 bonus to a saving throw or skill check of your choice (this choice must be related to the dream’s subject matter) and can reroll a failed check of that type once per day</p><h2>@UUID[Compendium.sfrpg.class-features.Item.HqBLb5H1BN81uHuB]{Visitation (Su) - 18th Level}</h2><p>Once per day, you can spend 1 Resolve Point as a full action to physically travel to the location of a specific dreamer through dreamspace. The target dreamer can be anywhere in the universe, including on another plane, as long as you’re familiar with them. The visitation lasts for a number of minutes equal to [[floor( @classes.mystic.levels / 2 )]]{one-half your character level}.</p>"
```
[the remark this answers is not in the document]
Comment (rmorgens) [reply src/items/class-features/dreamer.json]: Yeah, forgot to clean up the index file.  Hopefully fixed now

File: src/items/class-features/fortified_imagination_(su).json [hunk 12]
```diff
@@ -0,0 +1,143 @@
...
+    "damage": {
+      "parts": []
+    },
+    "damageNotes": "",
+    "description": {
+      "chat": "",
+      "gmnotes": "",
+      "short": "",
+      "unidentified": "",
+      "value": "<p>&gt;@UUID[Compendium.sfrpg.class-features.Item.udtZxolGt5mMq9SM]{Dreamer}</p>\n<p>You’ve experienced countless nightmares, preparing you to handle anything. You gain a +2 insight bonus to @Check[type:will] saves against mind-affecting spells and abilities. As a swift action, you can spend 1 Resolve Point to grant this bonus to an ally who can perceive you.</p>"
```
[the remark this answers is not in the document]
Comment (rmorgens) [reply src/items/class-features/fortified_imagination_(su).json]: Hopefully removed

**Comment by rmorgens** [comment]
Only Connection, Dreamer, Fortify Imagination, Visitation, Perceive Dreamspace and the related 'Dream ...' files should be in the PR.  Not sure how the others bled in, but should not be part of this.

File: src/items/class-features/dreamer.json (hunk 9 above)
Comment (ian612) [reply src/items/class-features/dreamer.json]: I think as long as the will save buttons have been removed and the formulas are spelled out in the text, it's okay to leave the automated DC calculations.

**Pull request merged**